### PR TITLE
Sc103415/test instructions

### PR DESCRIPTION
**Pull request by leah-is-offline**
sc103415

added instructions for how to test this action against itself and how other repositories can test against different version of this action (s/o nick)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,3 +93,28 @@ The action expects to receive a response to this message that looks like this:
...
+1. In build.yml make sure the `uses` clause in the `Upload coverage` step is set to `./`
+1. Open a pull request / draft pull request to trigger this workflow run. The code on your branch will be ran for the `Upload coverage` step upon github actions running `build.yml`
+
+#### Testing changes you made to another repository against this action
+
+1. In your repository that references this action, change your workflows code from
+ `uses: neofinancial/coverage-upload-action@v1` to `uses: neofinancial/coverage-upload-action@commit-hash` where the commit hash is the version of code you want to test against.
+1. Run the workflow that references this action.
+
+
```
Comment (leah-is-offline) [inline README.md:118]: let me know from a diff perspective if this makes sense or could be grammatically ... better.

**Review by AD-Blue — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -93,3 +93,28 @@ The action expects to receive a response to this message that looks like this:
...
+##### Testing changes you made to this repository
+
+1. In build.yml make sure the `uses` clause in the `Upload coverage` step is set to `./`
+1. Open a pull request / draft pull request to trigger this workflow run. The code on your branch will be ran for the `Upload coverage` step upon github actions running `build.yml`
+
+#### Testing changes you made to another repository against this action
+
+1. In your repository that references this action, change your workflows code from
+ `uses: neofinancial/coverage-upload-action@v1` to `uses: neofinancial/coverage-upload-action@commit-hash` where the commit hash is the version of code you want to test against.
+1. Run the workflow that references this action.
```
Comment (AdamTranquilla) [inline README.md:116]:
```suggestion
##### Testing action changes directly

1. In [build.yml](./.github/workflows/build.yml#L30) set `Upload coverage` steps `uses` clause to `./`
1. Open a draft pull request, or Re-run the `Build` job on an existing pull request to start this workflow run, then check the `Upload coverage` steps output

#### Testing action against another repo

1. Change the repositories to this action (`Upload coverage` steps `uses`) to `neofinancial/coverage-upload-action@<commit-hash>` using the commit hash you wish to test
1. Rerun the workflow that references this action
```

**Review by AdamTranquilla — DISMISSED** [review]
Looks good. Got a copy change if you want it

File: README.md [hunk 3]
```diff
@@ -93,3 +93,28 @@ The action expects to receive a response to this message that looks like this:
...
+1. Update the version in `package.json`
+1. Update `CHANGELOG.md`
+1. Commit your changes to github and open a pull request
+
+### Testing
+
+##### Testing changes you made to this repository
+
+1. In build.yml make sure the `uses` clause in the `Upload coverage` step is set to `./`
+1. Open a pull request / draft pull request to trigger this workflow run. The code on your branch will be ran for the `Upload coverage` step upon github actions running `build.yml`
```
Comment (neo-anand-patel) [inline README.md:110]: Do we need to include anything about `coverageEndpoint` here that points to a ngrok tunnel where the service is running?

**Review by PeterHuang516 — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (rogersnick) [inline README.md:116]: I think it's important that it is the full commit hash, and not the [shortened version](https://github.blog/changelog/2021-01-21-github-actions-short-sha-deprecation/).

File: README.md (hunk 3 above)
Comment (leah-is-offline) [reply README.md:110]: good point! ill add that

File: README.md (hunk 2 above)
Comment (leah-is-offline) [reply README.md:116]: ty

**Review by neo-anand-patel — APPROVED** [review]
nice

**Review by AdamTranquilla — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -104,15 +104,16 @@ These values are the % change in coverage for a particular metric and will be di
 
 ### Testing
 
-##### Testing changes you made to this repository
+##### Testing action changes directly
 
-1. In build.yml make sure the `uses` clause in the `Upload coverage` step is set to `./`
-1. Open a pull request / draft pull request to trigger this workflow run. The code on your branch will be ran for the `Upload coverage` step upon github actions running `build.yml`
+1. In [build.yml](./.github/workflows/build.yml#L30) set `Upload coverage` steps `uses` clause to `./`
```
Comment (AD-Blue) [inline README.md:109]:
```suggestion
1. In [build.yml](./.github/workflows/build.yml#L30) set the `Upload coverage` step's `uses` clause to `./`
```

File: README.md [hunk 5]
```diff
@@ -104,15 +104,16 @@ These values are the % change in coverage for a particular metric and will be di
...
+1. Open a pull request to run the `build` workflow. The code on your branch will be ran for the `Upload coverage` step upon github actions running. Check the `Upload coverage` steps output for more details about your run.
```
Comment (AD-Blue) [inline README.md:110]:
```suggestion
1. Open a pull request to run the `build` workflow. The code on your branch will be ran for the `Upload coverage` step upon github actions running. Check the `Upload coverage` step's output for more details about your run.
```

**Review by rogersnick — APPROVED** [review]
I googled `s/o` -- thanks 🎉

**Review by rogersnick — DISMISSED** [review]

**Review by AdamTranquilla — DISMISSED** [review]

**Review by AD-Blue — APPROVED** [review]

**Review by neo-anand-patel — DISMISSED** [review]

**Review by PeterHuang516 — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -125,3 +125,29 @@ Whether or not a response is received the action will print out general informat
...
+
+#### Testing action against another repository
+
+1. In your repository that references this action, change your workflows code from
+ `uses: neofinancial/coverage-upload-action@v1` to `uses: neofinancial/coverage-upload-action@<commit-hash>` where the commit hash is the **full** commit hash you wish to test.
+1. If your other repository is running a local server you want to test changes against, make sure you change the `coverageEndpoint` in `build.yml` to an ngrok url that points towards your local server.
+1. Run the workflow that references this action.
+
+
+
```
Comment (iansu) [inline README.md:153]:
```suggestion
These values are the % change in coverage for a particular metric and will be displayed in the `difference` column in the comment posted by the action.

## Contributing

1. Test your changes (see below)
1. Update the version in `package.json`
1. Update `CHANGELOG.md`
1. Commit your changes and open a pull request

### Testing

##### Testing Action changes in this repo

1. In [build.yml](./.github/workflows/build.yml#L30) set the `Upload coverage` step's `uses` clause to `./`
1. Open a pull request to run the `build` workflow. The code on your branch will be run for the `Upload coverage` step. Check the `Upload coverage` step's output for more details about your run.

#### Testing Action against another repository

1. In your repository that references this action, change your workflow's code from
 `uses: neofinancial/coverage-upload-action@v1` to `uses: neofinancial/coverage-upload-action@<commit-hash>` where the commit hash is the **full** commit hash you wish to test.
1. If your other repository is running a local server you want to test changes against, make sure you change the `coverageEndpoint` in `build.yml` to an ngrok url that points towards your local server.
1. Run the workflow that references this action.

```

Comment (iansu) [reply README.md:153]: We should probably also mention the `ACTION_STEP_DEBUG` and `ACTION_RUNNER_DEBUG` settings

**Review by iansu — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (leah-is-offline) [reply README.md:153]: Ticket to clean up logging and add debug already exists . will sent link to dx squad

**Review by neo-anand-patel — APPROVED** [review]

**Review by AD-Blue — APPROVED** [review]

**Review by rogersnick — APPROVED** [review]

**Review by AdamTranquilla — CHANGES_REQUESTED** [review]
lgtm

**Review by AdamTranquilla — APPROVED** [review]

**Review by PeterHuang516 — APPROVED** [review]

**Review by iansu — APPROVED** [review]

**Pull request merged**